src/concrete/ml/quantization/base_quantized_op.py

```diff
@@ -933,39 +933,67 @@ def make_output_quant_parameters(
+                (compute the lsbs_to_remove).
+            rounding_operation_id (Optional[str]): The identifier for a specific rounding
+                operation in a quantized operation. Used to create and access the
+                lsbs_to_remove value in the dictionary. Defaults to None
 
         Returns:
             numpy.ndarray: The rounded array.
         """
 
-        # Rounding is applied only if specified by user
-        if self.rounding_threshold_bits is not None:
-            if calibrate_rounding:
-                assert_true(
-                    not isinstance(x, fhe.tracing.Tracer),
-                    "Can't compute lsbs_to_remove at compilation time.",
-                )
-                assert_true(
-                    self.lsbs_to_remove is None,
-                    "Rounding has already been calibrated.",
+        # Initialize lsbs_to_remove if not already set
+        if not hasattr(self, "lsbs_to_remove") or self.lsbs_to_remove is None:
+            self.lsbs_to_remove = {} if rounding_operation_id else 0
+
+        if self.rounding_threshold_bits is not None and calibrate_rounding:
+            # Compute lsbs_to_remove only when calibration is True
+            current_n_bits_accumulator = compute_bits_precision(x)
+            computed_lsbs_to_remove = current_n_bits_accumulator - self.rounding_threshold_bits
+
+            assert_true(
+                not isinstance(x, fhe.tracing.Tracer),
+                "Can't compute lsbs_to_remove at compilation time.",
+            )
+
+            # Key-based logic (dictionary)
+            if rounding_operation_id:
+                # mypy
+                assert isinstance(self.lsbs_to_remove, dict)
+                self.lsbs_to_remove[rounding_operation_id] = max(
+                    self.lsbs_to_remove.get(rounding_operation_id, 0), computed_lsbs_to_remove
                 )
 
-                current_n_bits_accumulator = compute_bits_precision(x)
-                self.lsbs_to_remove = current_n_bits_accumulator - self.rounding_threshold_bits
+            # No key (integer logic)
+            else:
+                # mypy
+                assert isinstance(self.lsbs_to_remove, int)
+                self.lsbs_to_remove = max(self.lsbs_to_remove, computed_lsbs_to_remove)
+
+        # Rounding logic
+        if isinstance(self.lsbs_to_remove, dict):
+            lsbs_value = (
+                self.lsbs_to_remove.get(rounding_operation_id, 0) if rounding_operation_id else 0
+            )
+        else:
+            lsbs_value = self.lsbs_to_remove
 
-            # mypy
-            assert self.lsbs_to_remove is not None
+        # mypy
+        assert isinstance(lsbs_value, int)
 
-            # Apply rounding if needed
-            if self.lsbs_to_remove > 0:
-                x = fhe.round_bit_pattern(x, lsbs_to_remove=self.lsbs_to_remove)
+        if lsbs_value > 0:
+            x = fhe.round_bit_pattern(x, lsbs_to_remove=lsbs_value)
```

Comment:
@jfrery, since you are manually computing the nbr of lsb to remove,  I'm wondering if you encounter an overflow.

Reply:
The overflow thing is computed afterward as far as I know. Here I do not handle it.